website/profile/views.py

```diff
@@ -112,11 +112,15 @@ def update_user(auth):
             raise HTTPError(httplib.FORBIDDEN)
 
         for address in removed_emails:
-            if address == user.username:
-                raise HTTPError(httplib.FORBIDDEN)
             if address in user.emails:
-                user.remove_email(address)
-            user.remove_unconfirmed_email(address)
+                try:
+                    user.remove_email(address)
+                except PermissionsError as e:
+                    raise HTTPError(httplib.FORBIDDEN, e.message)
```

Comment:
In order to display the error message on the HTML error page you need to pass the message into the `data` argument

```python
raise HTTPError(.., data={'message_long': e.message})
```